docs/BUILD.md

```diff
@@ -1,6 +1,56 @@
 # Building s2n-tls
 
-s2n-tls can be built as follows:
+To use s2n-tls, you must build the library from the source and then include it in your program.
+## Requirements
+
+s2n-tls supports and tests on **x86** and **arm** architectures.
+
+### System requirements
+
+* 20GB RAM available
+
+### Supported OS and Distributions: 
+
+| OS     | Distros |      |        |
+|--------|---------|------|--------|
+| Redhat | Fedora  | AL   |
+| debian | ubuntu  |
+| *BSD   | free    | open | darwin |
+
+s2n-tls does not support [Windows](https://github.com/aws/s2n-tls/issues/497).
+
+<!-- We may want to move in this direction:
+
+### Redhat
+
+#### Fedora
+
+| Version | Kernel |
+|---------|--------|
+| 37      | 6.0    |
+| 38      | 6.2    | -->
+
+
+### Software requirements
+
+Building s2n-tls requires:
+
+1. Git
+1. GCC or Clang
+1. CMake
+1. OpenSSL
```

Comment:
A libcrypto is required, but it doesn't have to be OpenSSL. Other libcryptos are also supported, like AWS-LC. Or is this list just meant to be what's required to run the build examples? If so it might be good to rename the list to make that clear.